src/prommis/nanofiltration/costing/diafiltration_cost_model.py

```diff
@@ -0,0 +1,406 @@
+#####################################################################################################
+# “PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
+# (“PrOMMiS”) initiative, and is copyright (c) 2023-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory, et al. All rights reserved.
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license information.
+#####################################################################################################
+"""
+Flowsheet costing block for diafiltration flowsheet model
+"""
+
+from pyomo.environ import Constraint, Expression, NonNegativeReals, Param, Var, units
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+from idaes.core import declare_process_block_class, register_idaes_currency_units
+from idaes.core.util.constants import Constants
+
+from prommis.nanofiltration.costing.diafiltration_cost_block import (
+    DiafiltrationCostingBlockData,
+)
+
+
+@declare_process_block_class("DiafiltrationCosting")
+class DiafiltrationCostingData(DiafiltrationCostingBlockData):
+    """
+    Costing block for the diafiltration flowsheet
+    """
+
+    def build_global_params(self):
+        # Register currency and conversion rates based on CE Index
+        register_idaes_currency_units()
+
+        # initialize the common global parameters
+        self._build_common_global_params()
+
+        # Set the base year for all costs
+        self.base_currency = units.USD_2021
+
+        # Set a base period for all operating costs
+        self.base_period = units.year
+
+        # the following global parameters are from the reference file
+        self.factor_total_investment = Var(
+            initialize=2,  # TODO: verify
+            domain=NonNegativeReals,
+            doc="Total investment factor [investment cost/equipment cost]",
+            units=units.dimensionless,
+        )
+        self.factor_maintenance_labor_chemical = Var(
+            initialize=0.03,  # TODO: verify
+            domain=NonNegativeReals,
+            doc="Maintenance-labor-chemical factor [fraction of investment cost/year]",
+            units=units.year**-1,
+        )
+        self.factor_capital_annualization = Var(
+            initialize=0.1,  # TODO: verify
+            domain=NonNegativeReals,
+            doc="Capital annualization factor [fraction of investment cost/year]",
+            units=units.year**-1,
+        )
```

Comment:
Have these been verified?

Reply:
Yes, I think these are okay assumptions for now. I will remove these comments.